src/IbanNet/Validation/Rules/QrIbanRule.cs

```diff
@@ -0,0 +1,35 @@
+﻿using IbanNet.Validation.Results;
+
+namespace IbanNet.Validation.Rules;
+
+/// <summary>
+/// Asserts that the IBAN is a valid QR IBAN.
+/// </summary>
+internal sealed class QrIbanRule : IIbanValidationRule
+{
+    private static readonly string[] ValidCountries = ["CH", "LI"];
+
+    /// <inheritdoc />
+    public ValidationRuleResult Validate(ValidationRuleContext context)
+    {
+        string qrIban = context.Value;
+
+        if (qrIban.Length < 9)
+        {
+            return new InvalidQrIbanResult();
+        }
+
+        string countryIdentifier = qrIban.Substring(0, 2);
```

Comment:
Since this rule is always executed after the default ruleset, the IBAN is considered valid (from the spec point of view). So you could just pull the country code from `context.Country!.TwoLetterISORegionName`. This avoids the need to substring it again and upper case it later.

Reply:
Thank you for your feedback. Since there was no validation rule for it, I think I missed that extension you mentioned. Now I pushed another commit, and it's basically a wrapper around the already existing extension.